auto_nag/scripts/crash_samll_volume.py

```diff
@@ -0,0 +1,188 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import TOP_CRASH_IDENTIFICATION_CRITERIA, Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class CrashSmallVolume(BzCleaner):
+    def __init__(self):
+        super().__init__()
+
+        topcrash = Topcrash(
+            criteria=self._adjust_topcrash_criteria(TOP_CRASH_IDENTIFICATION_CRITERIA)
+        )
+        self.topcrash_signatures = topcrash.get_signatures()
+        self.blocked_signatures = topcrash.get_blocked_signatures()
+
+    def description(self):
+        return "Bugs with small crash volume"
+
+    def columns(self):
+        return ["id", "summary", "severity", "deleted_keywords_count"]
+
+    def _get_last_topcrash_added(self, bug):
+        pass
+
+    def _adjust_topcrash_criteria(self, topcrash_criteria):
+        factor = 2
+        new_criteria = []
+        for criterion in topcrash_criteria:
+            criterion = {
+                **criterion,
+                "tc_limit": criterion["tc_limit"] * factor,
+            }
+            if "tc_startup_limit" in criterion:
+                criterion["tc_startup_limit"] = criterion["tc_startup_limit"] * factor
+
+            new_criteria.append(criterion)
+
+        return new_criteria
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        existing_keywords = {
+            keyword
+            for keyword in ("topcrash", "topcrash-startup")
+            if keyword in bug["keywords"]
+        }
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+
+        top_crash_signatures = [
+            signature
+            for signature in signatures
+            if signature in self.topcrash_signatures
+        ]
+
+        keep_topcrash_startup = any(
+            any(
+                criterion["is_startup"]
+                for criterion in self.topcrash_signatures[signature]
+            )
+            for signature in top_crash_signatures
+        )
+
+        keywords_to_remove = None
+        if not top_crash_signatures:
+            keywords_to_remove = existing_keywords - {"topcrash", "topcrash-startup"}
```

Comment:
I'm confused here, if we have a bug with `topcrash` keywords and it is not a topcrash anymore, wouldn't we end up with an empty `keywords_to_remove` here?

Reply:
Yes, nice catch. It should be an intersection, not a difference. Fixed in 6ebfb46bdebc2d62c06aab8b236552e68d4c4787 and improved in 59495a7e5c9540e0936813b8dc01288905c6bedf.